yt/convenience.py

```diff
@@ -5,119 +5,124 @@
-    :class:`yt.data_objects.static_output.Dataset` subclass.
+    Additional arguments, if any, are passed down to the return class.
+
+    Returns
+    -------
+    Dataset
+        If fn is a single path, create a Dataset from the appropriate subclass.
+
+    DatasetSeries (Dataset container class)
+        If fn is a glob pattern (i.e. containing wildcards '[]?!*'), create a series.
+
+    Raises
+    ------
+    OSError
+        If fn does not match any existing file or directory.
+
+    YTOutputNotIdentified
+        If fn matches existing file or directory with undetermined format.
     """
-    args = _sanitize_load_args(*args)
-    candidates = []
-    valid_file = []
-    for argno, arg in enumerate(args):
-        if isinstance(arg, str):
-            if os.path.exists(arg):
-                valid_file.append(True)
-            elif arg.startswith("http"):
-                valid_file.append(True)
-            else:
-                if os.path.exists(os.path.join(ytcfg.get("yt", "test_data_dir"), arg)):
-                    valid_file.append(True)
-                    args[argno] = os.path.join(ytcfg.get("yt", "test_data_dir"), arg)
-                else:
-                    valid_file.append(False)
-        else:
-            valid_file.append(False)
-    types_to_check = output_type_registry
-    if not any(valid_file):
-        try:
-            from yt.data_objects.time_series import DatasetSeries
-            ts = DatasetSeries(*args, **kwargs)
-            return ts
-        except (TypeError, OSError, YTOutputNotIdentified):
-            pass
-        # We check if either the first argument is a dict or list, in which
-        # case we try identifying candidates.
-        if len(args) > 0 and isinstance(args[0], (list, dict)):
-            # This fixes issues where it is assumed the first argument is a
-            # file
-            types_to_check = dict((n, v) for n, v in
-                    output_type_registry.items() if n.startswith("stream_"))
-            # Better way to do this is to override the output_type_registry
+    fn = os.path.expanduser(fn)
+
+    if any([wildcard in fn for wildcard in "[]?!*"]):
+        from yt.data_objects.time_series import DatasetSeries
+        return DatasetSeries(fn, *args, **kwargs)
+
+    if not (os.path.exists(fn) or fn.startswith("http")):
+        alt_fn = os.path.join(ytcfg.get("yt", "test_data_dir"), fn)
+        if os.path.exists(alt_fn):
+            fn = alt_fn
         else:
-            mylog.error("None of the arguments provided to load() is a valid file")
-            mylog.error("Please check that you have used a correct path")
-            raise YTOutputNotIdentified(args, kwargs)
-    for n, c in types_to_check.items():
-        if n is None: continue
-        if c._is_valid(*args, **kwargs): candidates.append(n)
-
-    # convert to classes
-    candidates = [output_type_registry[c] for c in candidates]
+            raise OSError("No such file or directory: %s" % fn)
```

Comment:
Thanks for this suggestion. I'm not a fan of exclamative points in error messages but I can not think of anything better myself for now. I'll come back to this if I can think of something !

Reply:
Also this would require to check that the data dir was defined. Otherwise this would print
```
 Also tried /does/not/exist/<fn> with no better luck, sorry!
```
which isn't exactly more clear than leaving it out 😅

Reply:
done !